satellite/satellitedb/bandwidthagreement.go

```diff
@@ -19,60 +19,82 @@ type bandwidthagreement struct {
 }
 
 func (b *bandwidthagreement) CreateAgreement(ctx context.Context, rba *pb.RenterBandwidthAllocation) error {
-	rbaBytes, err := proto.Marshal(rba)
-	if err != nil {
-		return err
-	}
 	expiration := time.Unix(rba.PayerAllocation.ExpirationUnixSec, 0)
-	_, err = b.db.Create_Bwagreement(
+	_, err := b.db.Create_Bwagreement(
 		ctx,
 		dbx.Bwagreement_Serialnum(rba.PayerAllocation.SerialNumber+rba.StorageNodeId.String()),
-		dbx.Bwagreement_Data(rbaBytes),
-		dbx.Bwagreement_StorageNode(rba.StorageNodeId.Bytes()),
+		dbx.Bwagreement_StorageNodeId(rba.StorageNodeId.Bytes()),
+		dbx.Bwagreement_UplinkId(rba.PayerAllocation.UplinkId.Bytes()),
 		dbx.Bwagreement_Action(int64(rba.PayerAllocation.Action)),
 		dbx.Bwagreement_Total(rba.Total),
 		dbx.Bwagreement_ExpiresAt(expiration),
 	)
 	return err
 }
 
-func (b *bandwidthagreement) GetAgreements(ctx context.Context) ([]bwagreement.Agreement, error) {
-	rows, err := b.db.All_Bwagreement(ctx)
+const uplinkSQL = `SELECT uplink_id, SUM(total), 
```

Comment:
Inline, no need for a separate constant, also rebind is missing.

Reply:
@egonelbre what do you mean by 'rebind is missing'?

Reply:
postgres doesn't support `?`, so you need to use `db.Rebind(query)` to rewrite it

Reply:
from what I can tell, both support ${/d} syntax.  not sure if Rebind() would be better in any way.

Reply:
yeah, rebind does the right thing